### PR TITLE
Refactor components to be more reusable and adhere to .btn layout.

### PR DESCRIPTION
**Pull request by cutandpastey**
This PR refactors the sass build as well as re-laying out the files within the project.  This has been done to maximise re-usability between projects but also to bring other components into line with the `btn` component introduced in https://github.com/guardian/tools-components/pull/6

### CODE REVIEW

File: bin/lib/build-sass.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+let pWriteFile  = Q.denodeify(fs.writeFile);
+let pMkdirP     = Q.denodeify(mkdirp);
+let pRenderSass = Q.denodeify(sass.render);
+
+export default () => {
+
+  //Q.async takes a generator
+  //when the generator yields a promise Q will wait for the promise to resolve
+  //and then call generator.next()
+  return Q.async(function* (){
```
Comment (cutandpastey) [inline bin/lib/build-sass.js:21]: @jamesgorrie Genuinely useful implementation of a generator.

File: bin/build-sass-cli.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env node
```
Comment (jamespamplin) [inline bin/build-sass-cli.js]: Do we need the `.js` extension on things in the bin?

File: bin/lib/build-sass.js (hunk 1 above)
Comment (jamespamplin) [inline bin/lib/build-sass.js:21]: Lovely :)

File: bin/build-sass-cli.js (hunk 2 above)
Comment (cutandpastey) [inline bin/build-sass-cli.js]: No. Would you like me to remove them?

File: bin/lib/build-sass.js (hunk 1 above)
Comment (cutandpastey) [inline bin/lib/build-sass.js:21]: Plus the try/catch handles the errors really nicely.

File: bin/build-sass-cli.js (hunk 2 above)
Comment (jamespamplin) [inline bin/build-sass-cli.js]: Yes please. Most bin scripts don't have extensions when they have hash bangs.

Comment (cutandpastey) [inline bin/build-sass-cli.js]: vim will cry :'(

**Comment by robinedman** [comment]
Some of the SASS mixins are connected to just one component: row is a part of Grid for example. Would it be better to nest it under the component?

**Comment by cutandpastey** [comment]
@robinedman No. Check the updates to the readme.

**Comment by robinedman** [comment]
Ok that's fine. row and column are not component names though. Maybe it'd be good to use a directory as a namespace: `sass/mixins/grid/row`?

**Comment by cutandpastey** [comment]
Okay, makes sense.

**Comment by cutandpastey** [comment]
@jamespamplin @robinedman all issues should have been addressed.

**Comment by robinedman** [comment]
Great work, looking forward to start creating some components! :+1:

**Comment by cutandpastey** [comment]
Snapshots mate! Snapshots...

**Comment by jamespamplin** [comment]
Not sure I like separating the code by technology rather than concerns. I liked how js and scss to do with icons was in the `lib/grid` dir. Now you have to search in lib and sass.

**Comment by cutandpastey** [comment]
Yeah, but it makes it 100% easier for distribution. This way you only need to add `composer-components/sass` to your `includePaths` not `composer-components/lib/{component-name}/{component-name}.scss` AND `composer-components/sass`

**Comment by jamespamplin** [comment]
Ah, rightyo.

**Pull request merged**